source/slang/slang-ir-autodiff.cpp

```diff
@@ -362,6 +382,47 @@ IRInst* AutoDiffSharedContext::findDifferentiableInterface()
     return nullptr;
 }
 
+IRStructType* AutoDiffSharedContext::findNullDifferentialStructType()
```

Comment:
This is the point where I think we should just make NullDifferentialType an IR opcode, and use `__intrinsic_type($(kIROp_NullDifferentialType))` on the stdlib  declaration.

Reply:
We can find anything via namehint decoration, it can be removed when obfuscation is on.